src/avt/Queries/Queries/avtXRayImageQuery.C

```diff
@@ -300,8 +342,18 @@ avtXRayImageQuery::SetInputParams(const MapNode &params)
     if (params.HasNumericEntry("output_ray_bounds"))
         SetOutputRayBounds(params.GetEntry("output_ray_bounds")->ToBool());
 
-    if (params.HasNumericEntry("family_files"))
-        SetFamilyFiles(params.GetEntry("family_files")->ToBool());
+    if (params.HasEntry("filename_scheme"))
+    {
+        if (params.GetEntry("filename_scheme")->TypeName() == "int")
```

Comment:
Even if we support enum in the C++ code here, that doesn't mean such support needs to extend to handling it also in yaml. There, I think we should use case-insensitve string (not int) only. And setting the correct enum value based on the string would occur here (or near here) in just one place. In fact, its concievable we could enhance our processing of yaml/json to convert strings there to C++ enum's here by adopting a particular convention in the naming of the yaml/json member names.

Reply:
I did integers for two reasons:
1) That is how the output type works and I wanted to mirror that functionality
2) I can use the val to index my array of allowed strings

Reply:
Ok. Enums are a bit wierd especially when you are talking about some persistent format like yaml or json.

I extended the json-c implementation to have enums...

https://github.com/LLNL/json-cwx/blob/30008dc17cd5f787dedd303c51367bb5a8885271/json-cwx/tests/foo.json#L7

Angle brackets (`<>`) introduce an enum and its basically a dict but where the dict has a *first* entry...the actual enum member's value. The remaining items are the list of all possible choices (e.g. the *enumeration). Thats the weird part...for an enum, the compiler or metadata somewhere needs to capture not just the currently assigned value but also the list of possible values. And, in my json-c impl. anyways, that list gets enumerated right next to the value even if it is the same enum is used multiple times in the json object.

More than anyone wanted to know on the subject I am sure.